client/src/api/fetcher.ts

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable @typescript-eslint/no-throw-literal */
+/* eslint-disable @typescript-eslint/ban-types */
+import type { SimpleError } from './responses';
+
+const resolveUrl = (
+  url: string,
+  _queryParams: Record<string, unknown> = {},
+  pathParams: Record<string, string> = {}
+) => {
+  return url.replace(/\{\w*\}/g, (key) => pathParams[key.slice(1, -1)]);
+};
+
+// Typed only the subset of the spec we actually use (to be able to build a simple mock)
+export type FetchImpl = (
```

Comment:
This should be probably moved elsewhere, potentially used in other parts of the SDK

Reply:
I'm wondering if can use maybe the types that the TypeScript language defines in the "dom" lib:

https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L18664

Reply:
This enforces our users to add ``DOM`` to their ``tsconfig`` which I wanted to avoid. Also providing a subset of the spec we actually use to be able to build simple mocks.